test_all.sh

```diff
@@ -33,14 +33,35 @@ if [ "$INPUT" = "-h" ] || [ "$#" -gt 1 ]; then
   exit 1
 fi
 
+WITH_TIMEOUT=""
+if [ -x "$(command -v timeout)" ]; then
+  WITH_TIMEOUT="timeout -s KILL 5s"
+elif [ -x "$(command -v gtimeout)" ]; then # MacOS from `brew install coreutils`
+  WITH_TIMEOUT="gtimeout -s KILL 5s"
+fi
+
 for impl in $(ls calculate_average_*.sh | sort); do
   noext="${impl%%.sh}"
   fork=${noext##calculate_average_}
 
-  if output=$(./test.sh "$fork" "$INPUT" 2>&1); then
+  # ./test.sh calls ./prepare_$fork.sh e.g. to build native image
+  # which may take some time.
+  # Here we run it upfront, assuming that prepare result is cached
+  # to avoid timeout due to long preparation.
```

Comment:
It should cache anyways until new `mvnw clean verify`. Currently there are two submissions that build native image and they put it into target/ which works as expeceted.

Of course second run for test_all.sh completes faster due to cached native image.